google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ReadWriteTransaction.java

```diff
@@ -337,7 +338,10 @@ public ApiFuture<ResultSet> executeQueryAsync(
       final ParsedStatement statement,
       final AnalyzeMode analyzeMode,
       final QueryOption... options) {
-    Preconditions.checkArgument(statement.isQuery(), "Statement is not a query");
+    Preconditions.checkArgument(
+        (statement.getType() == StatementType.QUERY)
+            || (statement.getType() == StatementType.UPDATE && statement.hasReturningClause()),
+        "Statement must be a query or DML with returning clause");
     checkValidTransaction();
```

Comment:
Can we add a couple of tests to https://github.com/googleapis/java-spanner/blob/main/google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/AbortedTest.java for DML statements with `RETURNING` clauses to ensure that it works correctly with the internal retry mechanism of the Connection API?

Reply:
Added the tests.